packages/flutter/lib/src/foundation/memory_allocations.dart

```diff
@@ -18,6 +18,39 @@ const bool _kMemoryAllocations = bool.fromEnvironment('flutter.memory_allocation
 /// `--dart-define=flutter.memory_allocations=true`.
 const bool kFlutterMemoryAllocationsEnabled = _kMemoryAllocations || kDebugMode;
 
+/// Create an object with exemption from release.
```

Comment:
Replace "release" with "dispose"/"disposal" everywhere? Unless there is a difference why not stay with the same terms for consistency?

Reply:
As explained in the doc, the objects are not exempted from disposal. If the exempted objects are garbage collected without disposal, it is still considered as violation. But it is ok that they are not disposed/GCed by the end of the testing (or in app lifetime).

Reply:
I also found this a bit confusing due to the similarity of dispose and release. Would something ugly but a little more specific like`allowNotDisposedOrGCInTest` be better?

Reply:
alternative even longer and uglier but more obvious name: `markAsSingletonNotExpectedToDisposedOrGcInTest`